src/decoder/decodeqrdata.ts

```diff
@@ -219,7 +219,7 @@ function decodeByteSegment(bits: BitStream, result: resultByteArray, count: numb
   for (var i = 0; i < count; i++) {
     readBytes[i] = bits.readBits(8);
   }
-  result.val = readBytes
+  Array.prototype.push.apply(result.val, readBytes);
```

Comment:
Ooh yea missed that `readBytes` was an array as well, my bad. 

In that case can we use `concat`? 

`result.val = result.val.concat(readBytes)`? 

I'm not a huge fan of digging into `prototype` unless we _have_ to, and pushing items from one array onto another one seems like something we should be able to do without it.